tests/unit/test_config_provider.py

```diff
@@ -363,6 +363,20 @@ def test_can_get_config_provider_non_chain_provider(self):
         self.assertIsInstance(provider, ConstantProvider)
         self.assertEqual(value, 'bar')
 
+    def test_deepcopy_preserves_overrides(self):
+        provider = ConstantProvider(100)
+        config_store = ConfigValueStore(mapping={'fake_variable': provider})
+        config_store.set_config_variable('fake_variable', 'foo')
```

Comment:
Let's make the value more self-descriptive here to make the test easier to follow. I'd recommend just making the value `'override-value'` instead of `'foo'`